### PR TITLE
Add self isolation data

### PR DESCRIPTION
**Pull request by owenpigott**
Added ingest for Self Isolation data.
- Pass in metadata for further reporting.
- Feature flag to disable on production.

Updated existing tests.

### CODE REVIEW

File: lib_src/lib/usecase/add_self_isolation_requests.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import datetime
+from ..helpers import parse_date_of_birth, case_note_needs_an_update, concatenate_address
```
Comment (kosiakkatrina) [inline lib_src/lib/usecase/add_self_isolation_requests.py:2]: nitpicky: I don't think we're using case_note_needs_an_update yet

File: lib_src/lib/usecase/add_self_isolation_requests.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                    "FirstName": row.Forename.capitalize() if row.Forename else '',
+                    "LastName": row.Surname.capitalize() if row.Surname else '',
+                    "DobDay": f'{dob_day}',
+                    "DobMonth": f'{dob_month}',
+                    "DobYear": f'{dob_year}',
+                    "ContactTelephoneNumber": row.Phone2,
+                    "ContactMobileNumber": row.Phone,
+                    "EmailAddress": row.Email,
+                    "CallbackRequired": True,
+                    "HelpNeeded": "Welfare Call",
```
Comment (kosiakkatrina) [inline lib_src/lib/usecase/add_self_isolation_requests.py:46]: Need to double check that this is how we've been saving it in the db so far (casing and all)

File: lib_src/lib/usecase/add_self_isolation_requests.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+from ..helpers import parse_date_of_birth, case_note_needs_an_update, concatenate_address
+
+
+class AddSelfIsolationRequests:
+    def __init__(self, create_help_request, here_to_help_api):
+        self.create_help_request = create_help_request
+        self.here_to_help_api = here_to_help_api
+
+    def execute(self, data_frame):
+        print("add")
```
Comment (Duslerke) [inline lib_src/lib/usecase/add_self_isolation_requests.py:11]: Is this meant to be logged?

File: lib_src/lib/usecase/add_self_isolation_requests.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                    "DobYear": f'{dob_year}',
+                    "ContactTelephoneNumber": row.Phone2,
+                    "ContactMobileNumber": row.Phone,
+                    "EmailAddress": row.Email,
+                    "CallbackRequired": True,
+                    "HelpNeeded": "Welfare Call",
+                    "NhsNumber": row['NHS Number'],
+                    "NhsCtasId": row.ID
+                }]
+            print(help_request)
```
Comment (kosiakkatrina) [inline lib_src/lib/usecase/add_self_isolation_requests.py:50]: Do we need this print?

Comment (Duslerke) [inline lib_src/lib/usecase/add_self_isolation_requests.py:50]: Are we sure we want to log this?

File: lib_src/lib/usecase/add_self_isolation_requests.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+                data_frame.at[index, 'help_request_id'] = help_request_id
+                data_frame.at[index, 'resident_id'] = resident_id
+
+                print(
+                    f'Added CEV {index + 1} of {len(data_frame)}: resident_id: {resident_id} help_request_id: {help_request_id}')
+
+        return data_frame
+
+    def is_self_isolation_request(self, row):
+        return True if row["LA Support Required"] == '1' or row["LA Support Letter Received"] == "1" else False
```
Comment (Duslerke) [inline lib_src/lib/usecase/add_self_isolation_requests.py:71]: I wonder if this (`return True if row["LA Support Required"] == '1' or row["LA Support Letter Received"] == "1" else False`) could be simplified to:

`return row["LA Support Required"] == '1' or row["LA Support Letter Received"] == "1"`

**Review by Duslerke — APPROVED** [review]
All good. Added couple comments

**Pull request merged**